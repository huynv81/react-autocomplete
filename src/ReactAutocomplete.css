.autoComplete {
  width: 100%;
  padding: 0.5rem;
  font-size: 1rem;
  outline: none;
}

.dropdownOptions {
  position: absolute;
  width: 100%;
  background-color: white;
  display: none;
  padding: 0;
  margin: 0;
  margin-top: -1px;
  z-index: 1000;
  overflow-x: hidden;
}

.dropdownOptions.open {
  display: block;
  box-shadow: 2px 2px 10px 0 rgba(150, 150, 150, 0.17);
}

.options {
  list-style: none;
  margin: 0;
  max-height: 10rem;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 0;
}

.option {
  cursor: pointer;
  font-size: inherit;
  color: #63666a;
  padding: 0.8em 1.5em;
}

.option:hover,
.option.selected {
  background-color: #cce2ff;
  color: black;
}
