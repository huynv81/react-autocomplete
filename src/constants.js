export const AUTOCOMPLETE_SUGGESTIONS = [
  "Alligator",
  "Bask",
  "Crocodilian",
  "Death Roll",
  "Eggs",
  "Jaws",
  "Reptile",
  "Solitary",
  "Tail",
  "Wetlands"
];
