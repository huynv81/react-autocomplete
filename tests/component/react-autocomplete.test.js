import Tester from '../Tester';
import ReactAutocomplete from '../../src/react-autocomplete';
import { AUTOCOMPLETE_OPTIONS } from '../../demo/constants';

const tester = new Tester();

const autocompleteOptions = AUTOCOMPLETE_OPTIONS.map(o => ({ text: o, value: o }));

describe('Autocomplete', () => {
  test('it should render', () => {
    const snapshot = tester.getSnapshot(
      ReactAutocomplete,
      { options: [] },
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('it should render with loading state', () => {
    const snapshot = tester.getSnapshot(
      ReactAutocomplete,
      { options: autocompleteOptions, loading: true },
    );
    expect(snapshot).toMatchSnapshot();
  });

  test('it should populate suggestions when value in the input changes', () => {
    const mockCallback = jest.fn();
    const testValue = 'Bas';
    const { component } = tester.getMountedInstance(
      ReactAutocomplete,
      { options: autocompleteOptions, onChange: mockCallback },
    );

    component.find('input').at(0).simulate('change', { target: { value: testValue } });

    const possibleMatches = autocompleteOptions.filter(o => o.text.toLowerCase().includes(testValue.toLowerCase()));
    expect(component.find('li').length).toEqual(possibleMatches.length);
    expect(mockCallback).toHaveBeenCalled();
  });

  test('it should invoke onSelectOption callback when user selects an option', () => {
    const testValue = 'Bas';
    const mockCallback = jest.fn();

    const { component } = tester.getMountedInstance(
      ReactAutocomplete,
      { options: autocompleteOptions, onSelectOption: mockCallback },
    );

    const input = component.find('input').at(0);
    input.simulate('change', { target: { value: testValue } });
    input.simulate('keydown', { key: 'ArrowDown' });
    input.simulate('keydown', { key: 'Enter' });

    expect(mockCallback).toHaveBeenCalled();
  });

  test('it should invoke onClearOptions callback when user selects an option', () => {
    const testValue = 'Bas';
    const mockCallback = jest.fn();

    const { component } = tester.getMountedInstance(
      ReactAutocomplete,
      { options: autocompleteOptions, onClearOptions: mockCallback },
    );

    const input = component.find('input').at(0);
    input.simulate('change', { target: { value: testValue } });
    input.simulate('keydown', { key: 'ArrowDown' });
    input.simulate('keydown', { key: 'Esc' });

    expect(mockCallback).toHaveBeenCalled();
  });
});
