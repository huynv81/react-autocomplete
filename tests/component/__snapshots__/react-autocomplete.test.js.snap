// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete it should render 1`] = `
Array [
  <input
    className="autoComplete"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
  />,
  <div
    className="dropdownOptions "
  />,
]
`;

exports[`Autocomplete it should render with loading state 1`] = `
Array [
  <input
    className="autoComplete"
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
  />,
  <div
    className="dropdownOptions open"
  >
    <ul
      className="options"
      role="listbox"
    >
      <li
        aria-selected={false}
        role="option"
      >
        Loading...
      </li>
    </ul>
  </div>,
]
`;
